lib/mongo/uri.rb

```diff
@@ -615,6 +626,94 @@ def auth_mech_props(value)
       properties
     end
 
+    # Parses the zlib compression level.
+    #
+    # @param value [ String ] The zlib compression level string.
+    #
+    # @return [ Integer | nil ] The compression level value if it is between -1 and 9 (inclusive),
+    #   otherwise nil (and a warning will be raised).
+    def zlib_compression_level(value)
+      if /^-?\d+$/ =~ value
```

Comment:
Ruby uses \A and \z for beginning and end of input. ^ and $ mean beginning and end of line.

Reply:
I'm not super worried about users embedding newlines into their URI options, but ok.

Reply:
Done